custom_components/hubitat/fan.py

```diff
@@ -0,0 +1,85 @@
+"""Support for Hubitat fans."""
+
+from logging import getLogger
+import re
+from typing import Any, List, Optional
+
+from hubitatmaker import (
+    ATTR_SPEED,
+    CAP_FAN_CONTROL,
+    CMD_SET_LEVEL,
+    CMD_OFF, 
+    CMD_ON, 
+    CMD_CYCLE_SPEED, 
+    CMD_SET_SPEED,
+    Device
+)
+
+from homeassistant.components.fan import (
+    FanEntity
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.util import color as color_util
+
+from .device import Hub, HubitatEntity, get_hub
+_LOGGER = getLogger(__name__)
+
+class HubitatFan(HubitatEntity, FanEntity):
+    """Representation of a Hubitat fan."""
+
+    @property
+    def is_on(self) -> bool:
+        """Return True if the switch is on."""
+        return self.get_str_attr("switch") == "on"
+
+    # @property
+    # def brightness(self) -> Optional[int]:
+    #     """Return the level of this fan."""
+    #     level = self.get_int_attr("level")
+    #     if level is None:
+    #         return None
+    #     return int(255 * level / 100)
+
+    @property
+    def speed(self) -> str:
+        """Return the speed of this fan."""
+        return self.get_str_attr(ATTR_SPEED)
+
+    @property
+    def speed_list(self) -> str:
+        """Return the list of speeds for this fan. (MakerAPI isn't sending the list of speeds)"""
+        return ["off", "low", "medium-low", "medium", "high", "auto"]
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn on the switch."""
+        _LOGGER.debug(f"Turning on {self.name} with {kwargs}")
```

Comment:
```suggestion
        _LOGGER.debug("Turning on %s with %s", self.name, kwargs)
```

Prefer the % placeholder format for logging statements (I didn't figure out that was the preferred way to go until a ways into the code).